### PR TITLE
refetch input definitions when an input value is updated

### PR DESCRIPTION
**Pull request by go-oleg**
One input's value can be used in a following input's options, so when a value changes, we need to refetch the input definitions via `/prepare` using the updated list of values.

A couple other changes were necessary to make this happen
 - push outriggerUrl into the inputs' store (so we can build the proper OutriggerAPI and call getInputs())
 - push the bundle into the inputs' store (required in the OutriggerAPI getInputs() call)

@mnibecker

### CODE REVIEW

**Comment by go-oleg** [comment]
cc @rlgomes, add a test.

File: test/inputs/actions.spec.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    items: [{value: 1, label: "one"}, {value: 2, label: "new value for as"}]
+                }
+            }
+        ];
+
+        const action = {
+            type: "INPUT_DEFS_UPDATE",
+            payload: mockResponseBody
+        };
+        const expectedActions = [action];
```
Comment (mnibecker) [inline test/inputs/actions.spec.js]: hows about:
```
const expectedActions = [{
   type: "INPUT_DEFS_UPDATE",
   payload: mockResponseBody
}]
```

File: src/inputs/index.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ export default class Input {
 
         this.store = store;
 
+        store.dispatch(ActionCreators.updateOutriggerUrl(`http://${outriggerUrl}`));
```
Comment (mnibecker) [inline src/inputs/index.js]: idk, the fact that we're setting outriggerUrl into two different places, doesn't sit well with me. If this changes, someone will have to know that have to set it twice. At bare minimum we should probably be instantiate `OutriggerAPI` with `store.getState().outriggerUrl`, but would love to find a better solution.

**Comment by mnibecker** [comment]
Should probably create/update the `CHANGELOG.md` on this one.

File: src/inputs/actions.js [hunk 3]
```diff
@@ -1,17 +1,55 @@
...
+    return (dispatch, getState) => {
+        let currentInput = _.findWhere(getState().inputs, { id: input_id });
+
+        if (currentInput.value === value) {
+            return;
+        }
+
+        let api = new OutriggerAPI(getState().outriggerUrl);
+        api.getInputs(getState().bundle, Object.assign({}, getValuesFromInputs(getState().inputs), {
+            [input_id]: value
```
Comment (mnibecker) [inline src/inputs/actions.js]: did not know you could do this

File: test/inputs/actions.spec.js (hunk 1 above)
Comment (mnibecker) [inline test/inputs/actions.spec.js]: well, I guess this what arnaudbenard does, is it bad if its not a `const`?

**Comment by go-oleg** [comment]
@mnibecker, implemented your suggestions (but didn't backfill the release history)

File: test/inputs/actions.spec.js (hunk 1 above)
Comment (go-oleg) [inline test/inputs/actions.spec.js]: Changed, more a relic of my copying the example from redux-mock-store than intentionally doing it.

**Comment by mnibecker** [comment]
yeah, definitely not expecting you to backfill release history  +1

**Pull request merged**